request.js

```diff
@@ -0,0 +1,42 @@
+var instagramPosts = require('./posts.js')
+var https = require('https')
+
+function fetchImages(instagramPosts, callback) {
+  var instagramImages = []
+  var counter = 0
+  var numberOfPosts = instagramPosts.length
+  instagramPosts.forEach(function(post, index){
+    var timeStart = Date.now()
+    https.get(post, function(response) {
+      var body = ''
+      response.on('data', function(chunk) {
+        body += chunk
+      })
+      response.on('end', function () {
+        var timeEnd = Date.now()
+        var needle = '<meta property="og:image" content="'
+        var urlStart = body.indexOf(needle)
+        var urlEnd = body.indexOf('?', urlStart + needle.length)
+        instagramImages[index] = body.substring(urlStart+needle.length, urlEnd)
+        counter++
+        if (require.main === module) console.log('Request with index ' + index + ' took', timeEnd-timeStart, 'ms')
```

Comment:
I think we should be careful about the style that we show students in the solutions. One-line `if` blocks are probably okay for quick, short-circuiting returns; but this line gets a bit long and should probably be split into two lines.